From 1852452c0682edccdafa9b22d02fc9d0bc2236b5 Mon Sep 17 00:00:00 2001
From: Quanyang Wang <quanyang.wang@windriver.com>
Date: Wed, 15 May 2019 18:21:02 +0800
Subject: [PATCH 08/22] driver: clk: imx: add IMX8MQ_CLK_OCOTP_ROOT to
 clks_init_on array

By default, the clock "ocotp_root_clk" is enabled by u-boot. But
when running kexec, this clock is disabled and it will result system
hang when reading register OCOTP_UID_HIGH. So add it to clks_init_on.

Signed-off-by: Quanyang Wang <quanyang.wang@windriver.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx8mq.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/clk/imx/clk-imx8mq.c b/drivers/clk/imx/clk-imx8mq.c
index 9af1f94f3ce6..0616cbdb65a5 100644
--- a/drivers/clk/imx/clk-imx8mq.c
+++ b/drivers/clk/imx/clk-imx8mq.c
@@ -277,6 +277,7 @@ static int const clks_init_on[] __initconst = {
 	IMX8MQ_CLK_MAIN_AXI, IMX8MQ_CLK_A53_CG,
 	IMX8MQ_CLK_TMU_ROOT,
 	IMX8MQ_CLK_DRAM_APB, IMX8MQ_DRAM_PLL_OUT,
+	IMX8MQ_CLK_OCOTP_ROOT,
 };
 
 static struct clk ** const uart_clks[] __initconst = {
-- 
2.17.1

